src/termio/shell_integration.zig

```diff
@@ -273,11 +273,6 @@ fn setupBash(
     } else return null;
     try args.append(alloc, "--posix");
 
-    // On macOS, we request a login shell to match that platform's norms.
-    if (comptime builtin.target.os.tag.isDarwin()) {
-        try args.append(alloc, "--login");
```

Comment:
An alternate solution to just the `$0` issue would be to prefix `args[0]` with a `-` here, to complement the `--login`, but I worry that goes further down the "patch the problems as they come up" path.